aten/src/ATen/native/mps/kernels/UnfoldBackward.metal

```diff
@@ -52,7 +32,7 @@ kernel void unfold_backward(
   auto size = dim_size_step_ndim.y;
   auto step = dim_size_step_ndim.z;
   auto ndim = dim_size_step_ndim.w;
-  long pos[16];
+  long pos[max_ndim];
```

Comment:
Doesn't metal support templated arrays? Any reason we are not using them here? Thought it was something like:
```suggestion
  array<long,max_ndim> pos;
```

Reply:
Alas, not it does not seem to work
```
In file included from /Users/malfet/git/pytorch/pytorch/aten/src/ATen/native/mps/kernels/Indexing.metal:1:
/Users/malfet/git/pytorch/pytorch/c10/metal/indexing.h:39:3: error: no matching function for call to 'pos_from_thread_index'
  pos_from_thread_index(idx, pos, sizes, ndim);
  ^~~~~~~~~~~~~~~~~~~~~
/Users/malfet/git/pytorch/pytorch/c10/metal/indexing.h:22:13: note: candidate function not viable: no known conversion from '::metal::array<long, max_ndim>' to 'long *' for 2nd argument
```

Reply:
@malfet I see. You might need to unbox it with `pos.data()` or something, but yeah.